icu4j/main/classes/core/src/com/ibm/icu/impl/number/MixedUnitLongNameHandler.java

```diff
@@ -61,6 +61,9 @@ public static MixedUnitLongNameHandler forMeasureUnit(ULocale locale,
                                                           String unitDisplayName,
                                                           PluralRules rules,
                                                           MicroPropsGenerator parent) {
+        // In Java we rely on the assert. In ICU4C, where we can efficiently
```

Comment:
What does "we rely on the assert" mean? assert is mostly ignored at runtime. 

From https://docs.oracle.com/javase/7/docs/technotes/guides/language/assert.html:
""By default, assertions are disabled at runtime.""

Since this is a public method anyone can call, assert is simply the wrong thing to use here. An assert means "internal error in my code", not "someone called a method with unexpected arguments".

Reply:
ICU4J has public methods that are for internal use only. Additionally, classes that fall within the "impl" directories are also considered internal-use-only. (In public API classes, we mark internal-use-only methods with @Deprecated.)

I'm using an assert because I do want it ignored for production code - getComplexity() can be expensive, if it is a built-in unit, the unit's identifier has to be parsed.
 
 "assert false" does fail for "ant check", so works for my purposes.

Reply:
As in: calling MixedUnitLongNameHandler with anything other than a mixed unit is indeed considered an internal error in ICU4J code.

It would have been very nice if MeasureUnit was a class hierarchy. Unfortunately it isn't. :-/

Reply:
I've tried to improve this error message. Does it look better now?